As a user, I can run a single stably-named Ansible role

mikedep333 opened the pull request:
to install Pulp: pulp_all_services

Implementation includes:

1. The new role pulp_all_services, does nothing but depend.
2. pulp role split into pulp_common and pulp_api.
3. Removal of var pulp_install_api_service (due to it now being
controlled by the inclusion of the pulp_api service.)
4. A new list of roles for users to run if they wish to install
Pulp, starting with pulp_redis & pulp_database, then the Pulp
services.
5. All Pulp service roles & pulp_webserver (temporarily) declare
their existing dependency on pulp_common.
6. Many incorrect docs updates were fixed.
7. Fix misc unrelated yamllint errors.

mikedep333 commented:
The last force push incorporates #339

fao89 commented on docs/index.md:
```diff
@@ -139,7 +139,7 @@ Roles
 
-- [pulp](roles/pulp/): installs Pulp 3 from PyPi or source and provides basic config.
+- [pulp_common](roles/pulp_common): installs Pulp 3 from PyPi or source and provides basic config.
 - [pulp_content](roles/pulp_content): install, configure, and set the state of pulp content app.
```
I think we can add `pulp_api` here

mikedep333 replied:
It's not showing this as outdated, but I fixed it.

fao89 commented on roles/pulp_all_services/README.md:
```diff
@@ -0,0 +1,6 @@
+
+A role to install all Pulp services on a single host.
+
+Currently, all it does is depend on the requires roles.
```
how about list the required roles here?

mikedep333 commented:
@fao89 Please merge once CI passes again.

fao89 commented:
when I tested I got:
```
502 Bad Gateway
nginx/1.18.0
```

Spredzy commented on docs/index.md:
```diff
@@ -139,7 +139,8 @@ Roles
 pulp_installer is equipped with the following roles:
 
-- [pulp](roles/pulp/): installs Pulp 3 from PyPi or source and provides basic config.
+- [pulp_common](roles/pulp_common): installs shared components of the Pulp 3 services from PyPi or source and provides basic config
```
:+1:

Spredzy commented on roles/pulp_api/tasks/main.yml:
```diff
@@ -11,4 +12,11 @@
       become: true
       notify: Restart pulpcore-api.service
 
+    - name: Set the pulpcore-api service state
```
I faced the following when trying the branch

```
TASK [pulp_api : Set the pulpcore-api service state] ***********************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "failure 1 during daemon-reload: Failed to execute operation: Interactive authentication required.\n"}
```

mikedep333 replied:
Fixed, I forgot `become: true`

Spredzy commented on roles/pulp_common/README.md:
```diff
@@ -1,7 +1,7 @@
 Pulp
```
s/Pulp/Pulp Common

mikedep333 commented:
> when I tested I got:
> 
> ```
> 502 Bad Gateway
> nginx/1.18.0
> ```

Did you run a specific molecule scenario without destroy? Or did you test using pulplift? If you tested using pulplift, we need to update the playbooks once this is merged for the new roles list.

fao89 commented:
> > when I tested I got:
> > ```
> > 502 Bad Gateway
> > nginx/1.18.0
> > ```
> 
> Did you run a specific molecule scenario without destroy? Or did you test using pulplift? If you tested using pulplift, we need to update the playbooks once this is merged for the new roles list. Otherwise, pulp-api will not run.
> 
> Also, I just fixed the task `Set the pulpcore-api service state`

I tested with pulplift and I added pulp_redis after pulp_database here:
https://github.com/pulp/pulplift/blob/master/playbooks/source-install.yml

mikedep333 commented:
> > > when I tested I got:
> > > ```
> > > 502 Bad Gateway
> > > nginx/1.18.0
> > > ```
> > 
> > 
> > Did you run a specific molecule scenario without destroy? Or did you test using pulplift? If you tested using pulplift, we need to update the playbooks once this is merged for the new roles list. Otherwise, pulp-api will not run.
> > Also, I just fixed the task `Set the pulpcore-api service state`
> 
> I tested with pulplift and I added pulp_redis after pulp_database here:
> https://github.com/pulp/pulplift/blob/master/playbooks/source-install.yml

The problem is that pulp-api role is not included. It is not running.

just run:
```
  roles:
    - pulp_all_services
    - pulp_devel
```
This is what playbooks/example-source/playbook.yml has.

The full list of roles one could specify is in molecule/source-dynamic/converge.yml

The change to pulplift is coming as soon as this is merged.

fao89 commented:
> > > > when I tested I got:
> > > > ```
> > > > 502 Bad Gateway
> > > > nginx/1.18.0
> > > > ```
> > > 
> > > 
> > > Did you run a specific molecule scenario without destroy? Or did you test using pulplift? If you tested using pulplift, we need to update the playbooks once this is merged for the new roles list. Otherwise, pulp-api will not run.
> > > Also, I just fixed the task `Set the pulpcore-api service state`
> > 
> > 
> > I tested with pulplift and I added pulp_redis after pulp_database here:
> > https://github.com/pulp/pulplift/blob/master/playbooks/source-install.yml
> 
> The problem is that pulp-api role is not included. It is not running.
> 
> just run:
> 
> ```
>   roles:
>     - pulp_all_services
>     - pulp_devel
> ```
> 
> This is what playbooks/example-source/playbook.yml has.
> 
> The full list of roles one could specify is in molecule/source-dynamic/converge.yml
> 
> The change to pulplift is coming as soon as this is merged.

I totally forgot about `pulp-api`, I will test it real quick

mikedep333 commented:
> it worked!

Thank you for explicitly testing it!

I tested it quickly via a molecule container:

`molecule destroy -s default`
`molecule test -s default --destroy=never`
`docker exec -it centos-8 /bin/bash`
